frontend/src/components/Charts/Bar.tsx

```diff
@@ -1,63 +1,62 @@
 import * as Plot from '@observablehq/plot';
-
+import type { BarChartMetadata } from '../../types/generated';
 import { Chart, type ChartProps } from './Main';
 
-export interface BarChartProps extends Omit<ChartProps, 'spec'> {
-    xField: string;
-    categoryField: string;
-    colorDomain: string[];
-    colorRange: string[];
-    textColor?: string;
+export interface BarChartProps {
+    data: unknown[];
+    metadata: BarChartMetadata;
 }
 
-export const BarChart = ({
-    data,
-    title,
-    xField,
-    categoryField,
-    hasLegend,
-    colorDomain,
-    colorRange,
-    textColor,
-}: BarChartProps) => {
-    type DataItem = Record<string, number | string>;
-    const typedData = data as DataItem[];
+const truncateLabel = (label: string, maxLength = 15): string => {
+    return label.length > maxLength ? label.slice(0, maxLength) + '...' : label;
+};
+
+const formatValue = (value: number): string => {
+    if (value >= 1e9) return `${(value / 1e9).toFixed(1)}B`;
+    if (value >= 1e6) return `${(value / 1e6).toFixed(1)}M`;
+    if (value >= 1e3) return `${(value / 1e3).toFixed(1)}K`;
+    return value.toString();
+};
```

Comment:
Is this something we'd want to have a backend tool do at some point? It's hard to know where to put the brains with this type of app, I'm finding.

Reply:
yeah.. not sure. I think this is going to have to be prompt engineered  if data-specific or just in the UI if it works across all data